File: MATTHEW/leetcode/AIcoding/1.Gemini-2.5/javascript_solutions/506-relative-ranks.js
var findRelativeRanks = function(score) {
    const n = score.length;
    const scoreToRankMap = new Map();
    const sortedScores = [...score].sort((a, b) => b - a);

    for (let i = 0; i < n; i++) {
        const currentScore = sortedScores[i];
        let rankString;
        if (i === 0) {
            rankString = "Gold Medal";
        } else if (i === 1) {
            rankString = "Silver Medal";
        } else if (i === 2) {
            rankString = "Bronze Medal";
        } else {
            rankString = (i + 1).toString();
        }
        scoreToRankMap.set(currentScore, rankString);
    }

    const result = new Array(n);
    for (let i = 0; i < n; i++) {
        result[i] = scoreToRankMap.get(score[i]);
    }

    return result;
};

File: MATTHEW/leetcode/AIcoding/4.GPT-4o/cpp_solutions/506-relative-ranks.cpp
#include <vector>
#include <string>
#include <algorithm>
#include <unordered_map>

using namespace std;

class Solution {
public:
    vector<string> findRelativeRanks(vector<int>& score) {
        int n = score.size();
        vector<pair<int, int>> scores(n);
        for (int i = 0; i < n; ++i) {
            scores[i] = {score[i], i};
        }
        sort(scores.rbegin(), scores.rend());

        vector<string> result(n);
        for (int i = 0; i < n; ++i) {
            if (i == 0) result[scores[i].second] = "Gold Medal";
            else if (i == 1) result[scores[i].second] = "Silver Medal";
            else if (i == 2) result[scores[i].second] = "Bronze Medal";
            else result[scores[i].second] = to_string(i + 1);
        }
        return result;
    }
};